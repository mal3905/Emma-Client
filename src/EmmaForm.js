import React from 'react'


export default function EmmaForm(props) {
  const { className, ...otherProps } = props
  return (
    <form
      className={['Emma-form', className].join(' ')}
      action='#'
      {...otherProps}
    />
  )
}