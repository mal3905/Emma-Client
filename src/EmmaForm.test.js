import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import EmmaForm from './EmmaForm'

describe(`EmmaForm component`, () => {
  const props = {
    className: 'test-class-name',
    children: <p>test children</p>,
    'data-other': 'test-other-prop'
  }

  it('renders a form.EmmaForm by default', () => {
    const wrapper = shallow(<EmmaForm />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders the EmmaForm given props', () => {
    const wrapper = shallow(<EmmaForm {...props} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
