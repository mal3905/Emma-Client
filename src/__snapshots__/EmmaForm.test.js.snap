// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmmaForm component renders a form.EmmaForm by default 1`] = `
<form
  action="#"
  className="Emma-form "
/>
`;

exports[`EmmaForm component renders the EmmaForm given props 1`] = `
<form
  action="#"
  className="Emma-form test-class-name"
  data-other="test-other-prop"
>
  <p>
    test children
  </p>
</form>
`;
